diff --git a/setup.py b/setup.py
index eefe10d..d933710 100644
--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,7 @@ from setuptools.command.build_ext import build_ext
 from distutils.errors import *
 import pickle
 
-__version__ = "0.8"
+__version__ = "0.8-1"
 
 # Disable hard links, otherwise building distributions fails on OS X
 try:
@@ -19,6 +19,11 @@ except:
 if getattr(sys, 'getwindowsversion', None):
     libraries = ['ws2_32', 'iphlpapi']
     def_macros = [('WIN32', 1)]
+elif getattr(sys, 'platform', None) == 'cygwin':
+    libraries = ['ws2_32', 'iphlpapi']
+    def_macros = [('WIN32', 1),
+                  ('USE_SYS_TYPES_FD_SET', 1),
+                  ('__USE_W32_SOCKETS', 1)]
 else:
     mos = getattr(sys, 'platform', None)
     libraries = []
